i18n/changed.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+"""
+Determine if the source translation files are up-to-date.
+"""
+from __future__ import print_function
+from subprocess import CalledProcessError
+
+from i18n import Runner
+from i18n.execute import execute
+
+
+class Changed(Runner):
+    """
+    Class used to check if the source translation files are up-to-date
+    """
+    def run(self, args):
+        """
+        Main entry point of script
+        """
+        changes_detected = self.detect_changes()
+        message = self.get_message(changes_detected)
+        print(message)
+        if changes_detected:
+            return 1
+        return 0
+
+    def detect_changes(self):
+        """
+        Detect if changes have been made to the msgid or msgstr lines in the translation files.
+
+        Returns:
+          boolean: True, if changes detected; otherwise, False.
+
+        Note:
+          This method requires ``git`` to be installed on the executing machine.
+        """
+        try:
+            execute('git diff --exit-code -G "^[msgid|msgstr]"')
+        except CalledProcessError as ex:
+            if ex.returncode == 1:
```

Comment:
We should not assume the non-zero return return code will be `1`. According to the docs, [`subprocess.check_call`](https://docs.python.org/2/library/subprocess.html#subprocess.check_call)(the function that ultimately gets called by `execute`), will raise `CalledProcessError` for *any* non-zero exit code. Given that we don't know all of the error codes of `git diff` and don't want to have to change our code if `git diff` changes in the future, we should fail whenever an exception is raised:

```python
try:
    execute('git diff --exit-code -G "^(msgid|msgstr)"')
    return False
except CalledProcessError:
    return True
```